modules/sso/config/launch/setup/74/added/launch/keycloak-spi.sh

```diff
@@ -37,12 +36,24 @@ function add_vault() {
   fi
 }
 
-# KEYCLOAK-8129 Set Keycloak server's hostname to 'request' by default if SSO_HOSTNAME not set
-function set_server_hostname_spi_to_request() {
+# KEYCLOAK-8129 Set Keycloak server's hostname to 'default' by default if SSO_HOSTNAME not set
+function set_server_hostname_spi_to_default() {
 
   if [ -z "${SSO_HOSTNAME}" ]; then
 
-    local -r hostname_spi="<spi name=\"hostname\"><default-provider>request</default-provider><provider name=\"fixed\" enabled=\"true\"><properties><property name=\"hostname\" value=\"localhost\"/><property name=\"httpPort\" value=\"-1\"/><property name=\"httpsPort\" value=\"-1\"/></properties></provider></spi>"
+    local frontend_url=''
+    if [ -n "${SSO_HOSTNAME_FRONTEND_URL:-$KEYCLOAK_FRONTEND_URL}" ]; then
+        frontend_url="<property name=\"frontendUrl\" value=\"${SSO_HOSTNAME_FRONTEND_URL:-$KEYCLOAK_FRONTEND_URL}\"/>"
+    fi
+    local admin_url=''
+    if [ -n "${SSO_HOSTNAME_ADMIN_URL:-$KEYCLOAK_ADMIN_URL}" ]; then
+        admin_url="<property name=\"adminUrl\" value=\"${SSO_HOSTNAME_ADMIN_URL:-$KEYCLOAK_ADMIN_URL}\"/>"
+    fi
+    local force_backend_url_to_frontend_url=''
+    if [ -n "${SSO_HOSTNAME_FORCE_BACKEND_URL_TO_FRONTENT_URL:-$KEYCLOAK_FORCE_BACKEND_URL_TO_FRONTENT_URL}" ]; then
```

Comment:
SSO_HOSTNAME_FORCE_BACKEND_URL_TO_FRONTENT_URL pretty crazy variable name. Not sure this is really needed as it's more a workaround in certain cases when proxy isn't configured properly. So I would leave this out.